src/main/java/wolox/training/model/User.java

```diff
@@ -24,21 +22,17 @@
     @GeneratedValue(strategy = GenerationType.AUTO)
     private long id;
 
-    @Column(nullable = false)
+    @NotNull
     private String username;
 
-    @Column(nullable = false)
+    @NotNull
     private String name;
 
-    @Column(nullable = false)
+    @NotNull
     private LocalDate birthdate;
 
-    @Column(nullable = false)
-    @ManyToMany(cascade = CascadeType.ALL)
-    @JoinTable(name = "users_book",
-        joinColumns = @JoinColumn(name = "book_id", referencedColumnName = "id"),
-        inverseJoinColumns = @JoinColumn(name = "users_id",
-            referencedColumnName = "id"))
+    @NotNull
+    @ManyToMany(cascade = {CascadeType.MERGE, CascadeType.PERSIST})
```

Comment:
Almost there :)
MERGE is totally right, but PERSIST isn't. Because for the cases where the Book does already exist, it'll try persisting again.
You can look at MERGE as the instruction to "merge" changes on the model back to the database.
But we're lacking the one that brings modifications in the database to the model. And that's our dear REFRESH :)

Reply:
Ok! Done!